### PR TITLE
Mal's branch radiation mutation event

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
Content "Event Bus for Radiation"

#### Purpose of change
Replacement for  #74602.  Second half of fix for #70035.

#### Describe the solution
Adds event bus in radiation mutation to hang an EOC off of.

### CODE REVIEW

File: src/event.cpp [hunk 1]
```diff
@@ -170,6 +171,7 @@ DEFINE_EVENT_FIELDS( character_learns_spell )
 DEFINE_EVENT_FIELDS( character_loses_effect )
 DEFINE_EVENT_FIELDS( character_melee_attacks_character )
 DEFINE_EVENT_FIELDS( character_melee_attacks_monster )
+DEFINE_EVENT_FIELDS( character_radioactively_mutates )
```
Comment (RenechCDDA) [inline src/event.cpp:174]:
```suggestion
```

Actually not necessary, since it's using the character spec.

**Comment by RenechCDDA** [comment]
Looks like there's some error in the json, not sure exactly where.

Also I missed adding an entry to the memorial logger. Will need to do it just like this (I can't send suggestions to a completely new file):
https://github.com/CleverRaven/Cataclysm-DDA/pull/74604/files#diff-b339610bcde7227efd831b0183737c7566f1953ee5cf3f0eee73dd0a76799506R1116

File: src/event.h [hunk 2]
```diff
@@ -186,7 +187,7 @@ struct event_spec_character_item {
     };
 };
 
-static_assert( static_cast<int>( event_type::num_event_types ) == 102,
+static_assert( static_cast<int>( event_type::num_event_types ) == 103,
```
Comment (Maleclypse) [inline src/event.h:190]:
```suggestion
static_assert( static_cast<int>( event_type::num_event_types ) == 104,
```

File: src/event.cpp [hunk 3]
```diff
@@ -141,7 +142,7 @@ DEFINE_EVENT_HELPER_FIELDS( event_spec_empty )
 DEFINE_EVENT_HELPER_FIELDS( event_spec_character )
 DEFINE_EVENT_HELPER_FIELDS( event_spec_character_item )
 
-static_assert( static_cast<int>( event_type::num_event_types ) == 102,
+static_assert( static_cast<int>( event_type::num_event_types ) == 103,
```
Comment (Maleclypse) [inline src/event.cpp:145]:
```suggestion
static_assert( static_cast<int>( event_type::num_event_types ) == 104,
```

File: data/json/effects_on_condition/mutation_eocs/mutation_effect_eocs.json [hunk 4]
```diff
@@ -366,6 +366,53 @@
       }
     ]
   },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_RADIATION_MUTATION",
+    "eoc_type": "EVENT",
+    "required_event": "character_radioactively_mutates",
+    "condition": { "or": [ { "u_is_character": true }, { "u_is_npc": true } ] },
```
Comment (Maleclypse) [inline data/json/effects_on_condition/mutation_eocs/mutation_effect_eocs.json:374]:
```suggestion
    "condition": "u_is_avatar",
```

File: data/json/effects_on_condition/mutation_eocs/mutation_effect_eocs.json [hunk 5]
```diff
@@ -366,6 +366,53 @@
       }
     ]
   },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_RADIATION_MUTATION",
+    "eoc_type": "EVENT",
+    "required_event": "character_radioactively_mutates",
+    "condition": "u_is_avatar",
```
Comment (Maleclypse) [inline data/json/effects_on_condition/mutation_eocs/mutation_effect_eocs.json:374]:
```suggestion
    "condition": "u_is_character",
```

**Pull request merged**